core/src/main/java/com/datastax/oss/driver/internal/core/channel/InFlightHandler.java

```diff
@@ -356,20 +350,27 @@ private void abortAllInFlight(DriverException cause) {
    */
   private void abortAllInFlight(DriverException cause, ResponseCallback ignore) {
     if (!inFlight.isEmpty()) {
-      // Clear the map now and iterate on a copy, in case one of the onFailure calls below recurses
-      // back into this method
-      Set<ResponseCallback> toAbort = ImmutableSet.copyOf(inFlight.values());
-      inFlight.clear();
-      for (ResponseCallback responseCallback : toAbort) {
+      for (ResponseCallback responseCallback : inFlight.values()) {
         if (responseCallback != ignore) {
-          responseCallback.onFailure(cause);
+          fail(responseCallback, cause);
         }
       }
+      inFlight.clear();
       // It's not necessary to release the stream ids, since we always call this method right before
       // closing the channel
     }
   }
 
+  private void fail(ResponseCallback callback, Throwable failure) {
+    try {
+      callback.onFailure(failure);
+    } catch (Throwable throwable) {
+      // Protect against unexpected errors. We don't have anywhere to report the error (since
+      // onFailure failed), so log as a last resort.
+      LOG.error("[{}] Unexpected error while failing {}", logPrefix, callback, throwable);
```

Comment:
The level was WARN in the few places where this was already done, but I raised to ERROR because it's a serious issue.